--- rate_ajax.bak.php	Sat May 25 17:26:49 2013
+++ rate_ajax.php	Sat Jun  1 12:54:27 2013
@@ -111,6 +111,7 @@
     $options->userid = $USER->id;
     $options->itemid = $itemid;
 
+    $options->use_specific_scale = $scaleid; // If this is one of our three special scales, deal with only the current one (don't combine all three)
     $rm->delete_ratings($options);
 }
 
@@ -140,6 +141,7 @@
 $ratingoptions->items = array($item);
 $ratingoptions->aggregate = $aggregationmethod;
 
+$ratingoptions->use_specific_scale_only = TRUE; // If this is one of our three special scales, deal with only the current one (don't combine all three)
 $items = $rm->get_ratings($ratingoptions);
 $firstrating = $items[0]->rating;
 
@@ -167,6 +169,13 @@
     $result->aggregate = $aggregatetoreturn;
     $result->count = $firstrating->count;
     $result->itemid = $itemid;
+
+    if ($scaleid == $CFG->scale_to_use_for_triple_rating ||
+        $scaleid == $CFG->scale_to_use_for_triple_rating_2 ||
+        $scaleid == $CFG->scale_to_use_for_triple_rating_3) {
+
+      $result->itemid = "{$itemid}scale{$scaleid}"; // ALAN: This is used by JavaScript (module.js) to match correct HTML id and then update it
+    }
 }
 
 echo json_encode($result);
